examples/ex18p.cpp

```diff
@@ -107,23 +108,56 @@ int main(int argc, char *argv[])
                   "Enable or disable GLVis visualization.");
    args.AddOption(&vis_steps, "-vs", "--visualization-steps",
                   "Visualize every n-th timestep.");
+   args.ParseCheck();
 
-   args.Parse();
-   if (!args.Good())
+   // 2. Read the mesh from the given mesh file.
+   Mesh mesh = mesh_file.empty() ? EulerMesh(problem) : Mesh(mesh_file);
+   const int dim = mesh.Dimension();
+   const int num_equations = dim + 2;
+
+   if (problem == 5)
    {
-      if (Mpi::Root()) { args.PrintUsage(cout); }
-      return 1;
+      mesh.Transform([](const Vector &x, Vector &y)
+      {
+         y = x;
+         y *= 0.5;
+      });
+   }
+   //    Refine the mesh to increase the resolution. In this example we do
+   //    'seq_ref_levels' of uniform refinement, where 'seq_ref_levels' is a
+   //    command-line parameter.
+   for (int lev = 0; lev < ser_ref_levels; lev++)
+   {
+      mesh.UniformRefinement();
    }
-   if (Mpi::Root()) { args.PrintOptions(cout); }
 
-   // 3. Read the mesh from the given mesh file. This example requires a 2D
-   //    periodic mesh, such as ../data/periodic-square.mesh.
-   Mesh mesh(mesh_file, 1, 1);
-   const int dim = mesh.Dimension();
+   if (numProcs > mesh.GetNE())
+   {
+      if (Mpi::Root())
+      {
+         mfem_warning(
+            "The number of processor is larger than the number of elements.\n"
+            "Refine serial meshes until the number of elements is large enough");
+      }
+      while (mesh.GetNE() < numProcs)
+      {
+         mesh.UniformRefinement();
+      }
+   }
```

Comment:
This shouldn't be needed. Everything should hopefully work if there are MPI ranks with empty mesh partitions. (And if not, it should be considered a bug that should be fixed)